tests/JoinArrayTest.php

```diff
@@ -13,7 +13,18 @@ class JoinArrayTest extends AtkPhpunit\TestCase
 {
     private function getInternalPersistenceData(Persistence\Array_ $db): array
     {
-        return $this->getProtected($db, 'data');
+        $data = [];
+        /** @var Persistence\Array_\Db\Table $table */
+        foreach ($this->getProtected($db, 'data') as $table) {
+            foreach ($table->getRows() as $row) {
+                $rowData = $row->getData();
+                $id = $rowData['id'];
```

Comment:
hard-coded `'id'` field name here?

Reply:
yes - it is in specific test class, `id` here is safe to be assumed, the data are newly not indexed by some specific ID.